src/plugins/graph/d3-types.ts

```diff
@@ -14,28 +16,40 @@ export type DotsElt = SVGSVGElement | null;
 //  unknown: type of data attached to parent element (none in this case)
 export type DotSelection = Selection<SVGCircleElement, CaseData, SVGSVGElement, unknown>;
 
-// selects all all g elements, which contain inner and outer circles
-export function selectGraphDots(svg: DotsElt): DotSelection | null {
+// selects all all g elements, which contain inner and outer circles, for a DataConfiguration
+export function selectGraphDots(svg: DotsElt, dataConfiguration: IDataConfigurationModel): DotSelection | null {
```

Comment:
As explained in more detail elsewhere, if you have a separate `dotsRef` per layer, then you shouldn't need to pass the data configuration as an argument to this function or the other `select...` functions below.